src/test/java/org/broadinstitute/hellbender/utils/pileup/ReadPileupUnitTest.java

```diff
@@ -75,6 +76,71 @@ public void testSplitByReadGroup() {
+        Assert.assertEquals(pileupSample1.size(), rgCoverage*2, "Wrong number of elements for sample1");
+        Assert.assertTrue( pileupSample1.getReadGroupIDs().contains("rg0"), "Pileup for sample1 should contain rg0");
+        Assert.assertTrue( pileupSample1.getReadGroupIDs().contains("rg1"), "Pileup for sample1 should contain rg1");
+        Assert.assertFalse(pileupSample1.getReadGroupIDs().contains("rg2"), "Pileup for sample1 shouldn't contain rg2");
+        Assert.assertFalse(pileupSample1.getReadGroupIDs().contains("rg3"), "Pileup for sample1 shouldn't contain rg3");
+        Assert.assertFalse(pileupSample1.getReadGroupIDs().contains(null),  "Pileup for sample1 shouldn't contain null read group");
+
+        // sample2
+        final ReadPileup pileupSample2 = pileup.getPileupForSample("sample2", header);
+        Assert.assertEquals(pileupSample2.size(), rgCoverage, "Wrong number of elements for sample2");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains("rg0"), "Pileup for sample2 shouldn't contain rg0");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains("rg1"), "Pileup for sample2 shouldn't contain rg1");
+        Assert.assertTrue( pileupSample2.getReadGroupIDs().contains("rg2"), "Pileup for sample2 should contain rg2");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains("rg3"), "Pileup for sample2 shouldn't contain rg3");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains(null),  "Pileup for sample2 shouldn't contain null read group");
+
+        // null sample
+        final ReadPileup pileupNull = pileup.getPileupForSample(null, header);
+        Assert.assertEquals(pileupNull.size(), rgCoverage*2, "Wrong number of elements for null sample");
+        Assert.assertFalse(pileupNull.getReadGroupIDs().contains("rg0"), "Pileup for null sample shouldn't contain rg0");
+        Assert.assertFalse(pileupNull.getReadGroupIDs().contains("rg1"), "Pileup for null sample shouldn't contain rg1");
+        Assert.assertFalse(pileupNull.getReadGroupIDs().contains("rg2"), "Pileup for null sample shouldn't contain rg2");
+        Assert.assertTrue( pileupNull.getReadGroupIDs().contains("rg3"), "Pileup for null sample should contain rg3");
+        Assert.assertTrue( pileupNull.getReadGroupIDs().contains(null),  "Pileup for null sample should contain null read group");
+
+    }
+
+    @Test(expectedExceptions = UserException.ReadMissingReadGroup.class)
+    public void testSplitBySampleMissingReadGroupException() throws Exception {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader(1, 1, 1000);
+        final GATKRead read = ArtificialReadUtils.createArtificialRead(header, "nullRG", 0, 1, 10);
+        final ReadPileup pileup = new ReadPileup(loc, Collections.singletonList(read), 1);
+        final Map<String, ReadPileup> error = pileup.splitBySample(header, null);
```

Comment:
\[FindBugs\] INFO: DLS: Dead store to error in org\.broadinstitute\.hellbender\.utils\.pileup\.ReadPileupUnitTest\.testSplitBySampleMissingReadGroupException\(\)